src/librbd/io/ImageDispatch.h

```diff
@@ -81,6 +81,10 @@ class ImageDispatch : public ImageDispatchInterface {
       DispatchResult* dispatch_result, Context** on_finish,
       Context* on_dispatched) override;
 
+  bool invalidate_cache(Context* on_finish) override {
+    return false;
```

Comment:
Nit: add a real implementation that passes the call down to the `ObjectDispatcher`

Reply:
Added.

Reply:
And so that I replace 'm_image_ctx->io_object_dispatcher->invalidate_cache' with 'm_image_ctx->io_image_dispatcher->invalidate_cache'.